src/main/java/io/nats/client/support/JwtUtils.java

```diff
@@ -0,0 +1,145 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.client.support;
+
+import io.nats.client.NKey;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Base64;
+
+import static io.nats.client.support.JsonUtils.beginJson;
+import static io.nats.client.support.JsonUtils.endJson;
+
+public class JwtUtils {
+    static class Nats implements JsonSerializable {
+        String issuerAccount;
+        String[] tags;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            JsonUtils.addField(sb, "issuer_account", issuerAccount);
+            JsonUtils.addStrings(sb, "tags", Arrays.asList(tags));
+            JsonUtils.addField(sb, "type", "user");
+            JsonUtils.addField(sb, "version", 2);
+            return endJson(sb).toString();
+        }
+    }
+
+    static class Claim implements JsonSerializable {
+        Duration exp;
+        long iat;
+        String iss;
+        String jti;
+        String name;
+        Nats nats;
+        String sub;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            if (exp != null && !exp.isZero() && !exp.isNegative()) {
+                long unixSeconds = exp.toMillis() / 1000;
+                JsonUtils.addField(sb, "exp", unixSeconds);
+            }
+            JsonUtils.addField(sb, "iat", iat);
+            JsonUtils.addFieldEvenEmpty(sb, "jti", jti);
+            JsonUtils.addField(sb, "iss", iss);
+            JsonUtils.addField(sb, "name", name);
+            JsonUtils.addField(sb, "nats", nats);
+            JsonUtils.addField(sb, "sub", sub);
+
+            return endJson(sb).toString();
+        }
+    }
+
+    static final String ENCODED_CLAIM_HEADER =
+            ToBase64Url("{\"typ\":\"JWT\", \"alg\":\"ed25519-nkey\"}");
+
+    static final String NATS_USER_JWT_FORMAT = "-----BEGIN NATS USER JWT-----\n" +
+            "%s\n" +
+            "------END NATS USER JWT------\n" +
+            "\n" +
+            "************************* IMPORTANT *************************\n" +
+            "    NKEY Seed printed below can be used to sign and prove identity.\n" +
+            "    NKEYs are sensitive and should be treated as secrets.\n" +
+            "\n" +
+            "-----BEGIN USER NKEY SEED-----\n" +
+            "%s\n" +
+            "------END USER NKEY SEED------\n" +
+            "\n" +
+            "*************************************************************";
+
+    public static String issueUserCreds(String accountSeed, String issuerAccount, String userSeed, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
```

Comment:
Agreed. It would be more clear if this took the `NKey accountSigningKey` and `NKey userKey`, and simply avoid the `String accountSeed` variant. In fact, having sensitive credentials as `String` type is not recommended:

https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords

If you feel this is really necessary, passing as `char[]` is preferred (although, it is trivial for users to call `NKey.fromSeed(char[])` API themselves, so I don't think it is buying us anything, and only adds to the API surface area to maintain.

Reply:
I just removed this method to avoid the char[] vs String issue.